x-pack/plugins/monitoring/server/plugin.ts

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { combineLatest } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { i18n } from '@kbn/i18n';
+import { has, get } from 'lodash';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { TelemetryCollectionManager } from 'src/legacy/core_plugins/telemetry/server/collection_manager';
+import { LOGGING_TAG, KIBANA_MONITORING_LOGGING_TAG } from '../common/constants';
+import {
+  Logger,
+  PluginInitializerContext,
+  RequestHandlerContext,
+  KibanaRequest,
+  KibanaResponseFactory,
+  CoreSetup,
+  ICustomClusterClient,
+  ElasticsearchConfig,
+  ElasticsearchConfigType,
+} from '../../../../src/core/server';
+import { MonitoringConfig } from './config';
+// @ts-ignore
+import { requireUIRoutes } from './routes';
+// @ts-ignore
+import { instantiateClient } from './es_client/instantiate_client';
+// @ts-ignore
+import { initBulkUploader, registerCollectors } from './kibana_monitoring';
+// @ts-ignore
+import { initInfraSource } from './lib/logs/init_infra_source';
+import { registerMonitoringCollection } from './telemetry_collection';
+import { XPackMainPlugin } from '../../../legacy/plugins/xpack_main/server/xpack_main';
+import { LicensingPluginSetup } from '../../licensing/server';
+import { PluginSetupContract } from '../../features/server';
+import { LicenseService } from './license_service';
+import { MonitoringLicenseService } from './types';
+
+export interface LegacyAPI {
+  xpackMain: XPackMainPlugin;
+  telemetryCollectionManager: TelemetryCollectionManager;
+  elasticsearch: any;
+  opsInterval: number;
+  getOSInfo: () => any;
+  hapiServer: any;
+  getServerStatus: () => string;
+  serverEvents: any;
+  infra: any;
+}
+
+interface PluginsSetup {
+  usageCollection: UsageCollectionSetup;
+  licensing: LicensingPluginSetup;
+  features: PluginSetupContract;
+}
+
+interface MonitoringCoreConfig {
+  get: (key: string) => string;
+}
+interface MonitoringCore {
+  config?: () => MonitoringCoreConfig;
+  log?: Logger;
+  route?: (options: any) => void;
+  xpackInfo?: any;
+}
+
+export class Plugin {
+  private readonly initializerContext: PluginInitializerContext;
+  private readonly log: Logger;
+  private readonly getLogger: (...scopes: string[]) => Logger;
+  private cluster?: ICustomClusterClient;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.initializerContext = initializerContext;
+    this.log = initializerContext.logger.get(LOGGING_TAG);
+    this.getLogger = (...scopes: string[]) => initializerContext.logger.get(LOGGING_TAG, ...scopes);
```

Comment:
Logger already has this functionality built-in https://github.com/elastic/kibana/blob/5abca85feba7383a345b741b0c119ea8764dde66/src/core/server/logging/logger.ts#L98

Reply:
Yes, but I wanted a way to let all monitoring code call `getLogger` and it's already prefixed with the `LOGGING_TAG`

Reply:
Is it not the same? 
https://github.com/elastic/kibana/blob/5abca85feba7383a345b741b0c119ea8764dde66/src/core/server/logging/logger.ts#L94-L95